frontend/Panel.js

```diff
@@ -239,6 +241,12 @@ class Panel extends React.Component {
     });
   }
 
+  componentWillUpdate() {
+    this.hasLaunchEditor =
```

Comment:
This expression might also need to be in `componentDidMount`, not sure about the lifecycle here.